src/App.tsx

```diff
@@ -15,49 +15,131 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
 
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
 
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
+// Use this function in the render method to prepare goods
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  // To avoid the original array mutation
+  const visibleGoods = [...goods];
 
+  if (sortType === SortType.ALPHABET) {
+    visibleGoods.sort();
+  } else if (sortType === SortType.LENGTH) {
+    visibleGoods.sort((a, b) => a.length - b.length);
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+  // Sort and reverse goods if needed
+
+  // eslint-disable-next-line no-console
+  console.log(sortType, isReversed);
+
+  return visibleGoods;
+}
+
+// DON'T save goods to the state
+// type State = {
+//   isReversed: boolean,
+//   sortType: SortType,
+// };
+
+export const App = () => {
+  const [reorderOptions, setReorderOptions] = useState<ReorderOptions>({
+    sortType: SortType.NONE,
+    isReversed: false,
+  });
```

Comment:
I think it would be much easier and cleaner to just create two separate states: 
```
[sortType, setSortType] ....
[isReserved, setIsReserved]...
```